providers/oidc.go

```diff
@@ -282,6 +282,9 @@ func (p *OIDCProvider) findClaimsFromIDToken(ctx context.Context, idToken *oidc.
 func (p *OIDCProvider) extractGroupsFromRawClaims(rawClaims map[string]interface{}) []string {
 	groups := []string{}
 
+	if group, ok := rawClaims[p.GroupsClaim].(string); ok {
```

Comment:
I wonder if this should be a switch on the type since we are having multiple options here, something like

```
switch rawClaims[p.GroupsClaim].(type) {
case []string:
  ...
case string:
  ...
}
```

I see we aren't doing this at the moment and are instead casting to a `[]interface{}`, perhaps @stefansedich tried this before and it didn't work? Any ideas?

Reply:
@JoelSpeed using case statement now and also added tests

Reply:
Looks like I missed some other tests. Looking into it

Reply:
I could've sworn there was some conversation on this `rawGroups, ok := rawClaims[p.GroupsClaim].([]interface{})` -- but I can't seem to find anything on the conversation history anymore.

Reply:
The switch implementation, while explicit, is much longer and duplicates most of the logic in the original